File: Apps/Codaxy.Dextop.Showcase/Demos/Grids/BufferedGrid.cs
using System;
using Codaxy.Dextop.Data;
using System.Linq;
using System.Collections.Generic;

namespace Codaxy.Dextop.Showcase.Demos.Grids
{
    [Demo("BufferedGrid",
        Title = "Buffered Grid",
		Description = "Ext JS 4 brand new grid supports infinite scrolling, which enables you to load any number of records into a grid without paging.",
        Path = "~/Demos/Grids"
    )]
    [LevelAdvanced]
    [TopicDextopGrid]
    [CategoryCodeSnippet]
    public class BufferedGridWindow : DextopWindow
    {
        public override void InitRemotable(DextopRemote remote, DextopConfig config)
        {
            base.InitRemotable(remote, config);
			var rand = new Random();
			data = new List<GridModel>();
			for (var i = 0; i<5000; i++)
				data.Add(new GridModel { Id = i, Name = "Item " + i, Value = rand.Next(1000, 2000) });
			Remote.AddStore("model", Read);
        }

		List<GridModel> data;

		DextopReadResult<GridModel> Read(DextopReadFilter filter)
		{
			return DextopReadResult.CreatePage(data.AsQueryable(), filter);
		}       

        [DextopModel]
        [DextopGrid]
        class GridModel
        {
            [DextopModelId]
            public int Id { get; set; }
            
            [DextopGridColumn(flex=1)]
            public String Name { get; set; }
            
            [DextopGridColumn(width = 70)]            
            public int Value { get; set; }            
        }
    }
}

File: Apps/Codaxy.Dextop.Showcase/Demos/Grids/BufferedGrid.js
Ext.ns('Showcase');

Ext.define('Showcase.demos.BufferedGridWindow', {
	extend: 'Dextop.Window',
	width: 500,
	height: 300,

	title: 'Buffered Grid',
	requires: 'Ext.grid.GridPanel',

	initComponent: function () {
		var store = this.remote.createStore('model', {
			autoLoad: true,
			buffered: true,			
			pageSize: 100,
			remoteSort: true
		});

		var columns = this.remote.createGridColumns('model');

		var grid = Ext.create('Ext.grid.GridPanel', {
			border: false,
			store: store,
			columns: columns,
			disableSelection: false //it works but selection is lost on scrolling
		});

		Ext.apply(this, {
			layout: 'fit',
			items: grid
		});

		this.callParent(arguments);
	}
});
